libs/product/src/selectors/composite-product/composite-product.selectors.ts

```diff
@@ -157,9 +161,71 @@ const createCompositeProductSelectors = (): DaffCompositeProductMemoizedSelector
 		}
 	);
 
+	/**
+	 * Selector for the minimum discounted price for required items of a composite product, depending on the current selection of item options.
+	 * This would be used for a composite product that has required items without default selections.
+	 */
+	//todo use optional chaining where possible.
+	const selectCompositeProductMinDiscountedPrice = createSelector(
+		selectProductEntities,
+		selectCompositeProductAppliedOptionsEntitiesState,
+		(products, appliedOptionsEntities, props) => {
+			const product = selectProduct.projector(products, { id: props.id });
+			if(product.type !== DaffProductTypeEnum.Composite) {
+				return undefined;
+			}
+			const appliedOptions = selectCompositeProductAppliedOptions.projector(products, appliedOptionsEntities, { id: props.id });
+			return (<DaffCompositeProduct>product).items.reduce((acc, item) => daffAdd(
+				acc, 
+				appliedOptions[item.id] ? 
+					daffMultiply(
+						daffSubtract(appliedOptions[item.id].price, appliedOptions[item.id].discount && appliedOptions[item.id].discount.amount), 
+						appliedOptions[item.id].quantity
+					) : 
+					getMinimumCompositeItemDiscountedPrice(item)
+			), daffSubtract(product.price, product.discount ? product.discount.amount : 0));
+		}
+	);
+
+	/**
+	 * Selector for the minimum discounted price for required items of a composite product, depending on the current selection of item options.
+	 * This would be used for a composite product that has required items without default selections.
+	 */
+	//todo use optional chaining where possible.
+	const selectCompositeProductMaxDiscountedPrice = createSelector(
+		selectProductEntities,
+		selectCompositeProductAppliedOptionsEntitiesState,
+		(products, appliedOptionsEntities, props) => {
+			const product = selectProduct.projector(products, { id: props.id });
+			if(product.type !== DaffProductTypeEnum.Composite) {
+				return undefined;
+			}
+			const appliedOptions = selectCompositeProductAppliedOptions.projector(products, appliedOptionsEntities, { id: props.id });
+			return (<DaffCompositeProduct>product).items.reduce((acc, item) => daffAdd(
+				acc, 
+				appliedOptions[item.id] ? 
+					daffMultiply(
+						daffSubtract(appliedOptions[item.id].price, appliedOptions[item.id].discount && appliedOptions[item.id].discount.amount),
+						appliedOptions[item.id].quantity
+					) : 
+					getMaximumCompositeItemDiscountedPrice(item)
+			), daffSubtract(product.price, product.discount ? product.discount.amount : 0));
```

Comment:
I think that it _probably_ shouldn't, because it might end up being a red herring for developers in the future, since the only time it is actually accurate is when you have a composite product with all optional items. On the other hand, these two (min and max) selectors should always be accurate in all use cases for selecting the min and max discounted prices that include required items.

Reply:
makes sense